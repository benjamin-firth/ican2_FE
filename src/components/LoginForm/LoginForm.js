import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { fetchData } from '../../utils/apiCalls';
import { loginCurrentUser } from '../../actions';
import Loader from '../Loader/Loader';
import './LoginForm.scss';

const LoginForm = () => {
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');
  const [isLoading, setIsLoading] = useState(false);
  const currentUser = useSelector(state => state.currentUser);

  window.scrollTo(0, 0);

  const enterEmail = (e) => {
    setError('');
    setEmail(e.target.value);
  }

  const enterPassword= (e) => {
    setError('');
    setPassword(e.target.value);
  }

  const login = (e) => {
    e.preventDefault();
    setIsLoading(true);
    if (!email.length || !password.length) {
      setError('Please be sure you have filled out all sections.');
      setIsLoading(false);
    } else if (!email.includes('@') || !email.includes('.')) {
      setError('Please enter a valid email address.');
      setIsLoading(false);
    } else {
      getUser()
      .then(data => {
        dispatch(loginCurrentUser(data.data.users));
        setIsLoading(false);
      })
      .catch(error => setError('That user does not exist. Please sign up!'))
    }
  }

  const getUser = () => {
    const body = {"query": "{users(email: \""+ email + "\") {id name email mentor profile {gender aboutMe image fieldOfInterest} mentorProfile {fieldOfKnowledge experienceLevel workDayQuestion enjoymentQuestion teachingPointsQuestion adviceQuestion} location {city state}}}"};

    return fetchData(body);
  };

  return (
    currentUser.name ? <Redirect to='myprofile' /> :
    isLoading ? <Loader message='loading your profile...'/> :
    <section className='login-page'>
      <form className='login-form'>
        <div key={1}>
          <p>EMAIL</p>
          <input type='text' onChange={(e) => enterEmail(e)}/>
        </div>
        <div key={2}>
          <p>PASSWORD</p>
          <input type='text' type='password' onChange={(e) => enterPassword(e)}/>
        </div>
        {error && <p className='error-msg'>{error}</p>}
        <button className='login-submit-button' onClick={(e) => login(e)}>enter</button>
      </form>
    </section>
  );
}

export default LoginForm;
