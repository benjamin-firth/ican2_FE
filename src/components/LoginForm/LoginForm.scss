@import '../../variables.scss';

.login-page {
  background-color: $primarycolor;
  height: 100vw;
  padding: 10vw;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.login-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 0;
  padding: 3vw;
  background-color: white;
  width: 40vw;

  p {
    font-family: $primaryfont;
    margin: 0;
    margin-top: 1vw;
    color: $secondarycolor;
    font-size: 1.2vw;
  }

  button {
    width: 50%;
    margin: .5vw;
    margin-top: 2vw;
    padding: 1vw;
    border: 5px solid $secondarycolor;
    color: $secondarycolor;
    font-size: 1.5vw;
    font-family: $secondaryfont;
    &:hover {
      background-color: $secondarycolor;
      color: white;
      cursor: pointer;
    }
  }

  input {
    border: 5px solid #C7D0D7;
    font-family: $primaryfont;
    font-size: 30px;
    height: 35px;
    width: 36vw;
  }

}
