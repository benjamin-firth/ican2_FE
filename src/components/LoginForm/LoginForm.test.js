import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

jest.mock("react-redux", () => ({
  useSelector: () => 'state',
  useDispatch: () => jest.fn()
}));

describe('LoginForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <LoginForm />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('email input', () => {
    expect(wrapper.find('#stateEmail').prop('value')).toEqual('');
    wrapper.find('#stateEmail').simulate('change', {target: {value: 'test@test.com'}});
    expect(wrapper.find('#stateEmail').prop('value')).toEqual('test@test.com');
  });

  it('password input', () => {
    expect(wrapper.find('#statePassword').prop('value')).toEqual('');
    wrapper.find('#statePassword').simulate('change', {target: {value: 'testPassword'}});
    expect(wrapper.find('#statePassword').prop('value')).toEqual('testPassword');
  });

  describe('login button', () => {
    it('should fire login', () => {
      wrapper.find('#stateEmail').simulate('change', {target: {value: 'kjhkjh'}});
      wrapper.find('#statePassword').simulate('change', {target: {value: 'kbjkb'}});
      wrapper.find('.login-submit-button').simulate('click', {preventDefault: () => true});

      expect(wrapper.find('.error-msg').prop('value')).toEqual(undefined);
    })
  })

});
