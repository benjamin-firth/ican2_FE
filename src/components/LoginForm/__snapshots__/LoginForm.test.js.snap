// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot 1`] = `
<section
  className="login-page"
>
  <form
    className="login-form"
  >
    <div
      key="1"
    >
      <p>
        EMAIL
      </p>
      <input
        id="stateEmail"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      key="2"
    >
      <p>
        PASSWORD
      </p>
      <input
        id="statePassword"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      className="login-submit-button"
      onClick={[Function]}
    >
      enter
    </button>
  </form>
</section>
`;
